File: 백준/2015/solution.swift
let NK = readLine()!.split(separator: " ").compactMap { Int($0) }
let N = NK[0]
let K = NK[1]

let numbers = readLine()!.split(separator: " ").compactMap { Int($0) }
var prefix: [Int] = [numbers[0]]

for i in 1..<numbers.count {
    prefix.append(prefix.last! + numbers[i])
}

var answer = 0

var counter: [Int: Int] = [:]

for i in 0..<prefix.count {
    if prefix[i] == K {
        answer += 1
    }
    if counter[prefix[i]-K] != nil {
        answer += counter[prefix[i] - K]!
    } else {
        counter[prefix[i]-K] = 0
    }
    if counter[prefix[i]] == nil {
        counter[prefix[i]] = 0
    } else {
        counter[prefix[i]]! += 1
    }
}

print(answer)

File: 백준/2015/solution.py
from collections import defaultdict
n, k = map(int, input().split())
numbers = list(map(int, input().split()))

answer = 0
dic = defaultdict(int)
for i in range(1, len(numbers)):
    numbers[i] += numbers[i-1]
    
for i in range(len(numbers)):
    if numbers[i] == k:
        answer += 1
    answer += dic[numbers[i]-k]
    dic[numbers[i]] += 1
print(answer)